spec/support/preload_values.rb

```diff
@@ -0,0 +1,51 @@
+#
+# Sometimes, it is not necessary to know the exact number of queries it takes to preload a record
+# Instead it is only necessary to know that the values actually do get preloaded
+#
+# This takes a relation or array as a block or argument.
+# It runs order by id if it is not already ordered
+# It loads the relation if necessary (by running load)
+#
+# It then ensures the correct values, and in the process counts queries to ensure they were preloaded
+#
+# If a single value is passed it, it assumes all records have the same value
+#
+# Example usage:
+#   expect(MyModel.includes(:relation)).to preload_values(:attribute, values)
```

Comment:
A few things like this:

I think it would help having two examples:  One with an array and one with a single value:

```ruby
expect(MyModel.includes(:relation)).to preload_values(:attribute, "foo")
expect(MyModel.includes(:relation)).to preload_values(:attribute, %w[foo bar baz])
```

Second, your example uses the "non-block" form, but you never actually use it in the form in the specs.  My question here is actually:  does this make sense to support both forms?  Seems like it might be simpler to pick one and go with that.  Personally I think the non-block is easier to type out, but I don't think it matters one way or the other.

Reply:
Yea, at first, the code did more. it counted queries before and stuff like that.

Now, the paren version works just as well and is easier to read.
may just circle back and change this in bulk rather than try and fix the slew of commits (with cascading git merge failures)